textual_autocomplete/_autocomplete.py

```diff
@@ -368,6 +366,12 @@ def close(self) -> None:
         if self.display:
             self.display = False
 
+    def on_mouse_move(self, event: events.MouseMove) -> None:
+        self.child.selected_index = event.y
```

Comment:
I suspect this will need to account for scrolling by adding on scroll_offset.y

Reply:
I've tested this with both scroll on the dropdown and scroll on the page - unaffected by `scroll_offset`.

If it's okay - I have a few other changes I can push to another pull request once I've tidied them up.

* allows styles to be passed in for `highlight_ranges` so you can group options in the dropdown. I use this to differentiate selections in the list.
* some logic changes. Adds a `DropdownSelected` event instead of bubbling the `Input.Selected` on dropdown item select
* I also have a preventable input that I find useful - can disable the user from being able to type (refreshes the cursor blink only) I use it in conjunction with autocomplete, (could also use it to limit input length etc) However that's more for textual I think. (not sure if others would find that useful)

Can also use this as a context manager:

```python
with my_input.prevent_input():
    ...
    # do stuff here where the user can't type
```

Scroll on selection:
<img width="999" alt="scroll_on_selection" src="https://github.com/darrenburns/textual-autocomplete/assets/22514713/8c7262a6-0512-455e-83a2-5b796dcb0077">

Selection made:
<img width="1004" alt="selection_made" src="https://github.com/darrenburns/textual-autocomplete/assets/22514713/b76bb9de-3c40-4550-9c93-0abd11a08682">

This is for highlight ranges, I use it to show which repo's are private (green):

<img width="982" alt="highlight_ranges" src="https://github.com/darrenburns/textual-autocomplete/assets/22514713/2585007b-45a8-4549-9d34-6e8ebd5b3ef8">

Cheers